src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -120,7 +120,8 @@ Newline5: NEWLINE
 Leading_asterisk3: LEADING_ASTERISK -> type(LEADING_ASTERISK);
 XmlTagOpen1: '<' -> type(OPEN), pushMode(xmlTagDefinition);
 STRING: '"' .*? '"' {referenceCatched = false;} -> mode(DEFAULT_MODE);
-PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {referenceCatched = true;};
+PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {_input.LA(1) == '.' || _input.LA(1) == '}'
+                     || Character.isWhitespace(_input.LA(1))}? {referenceCatched = true;};
```

Comment:
Consider this input:

```
@see https://www.something.com
```

Just for such cases I have been forced to use `_input.LA(1) == '.' || ...` for `PACKAGE` token. If I don't do that `https` gets treated as `PACKAGE` token. Following is the tree produced by my checkstyle master for the above input:

```
JAVADOC -> JAVADOC [0:0]
|--LEADING_ASTERISK -> * [0:0]
|--WS ->   [0:1]
|--JAVADOC_TAG -> JAVADOC_TAG [0:2]
|   |--SEE_LITERAL -> @see [0:2]
|   |--WS ->   [0:6]
|   |--REFERENCE -> REFERENCE [0:7]
|   |   `--PACKAGE -> https [0:7]
|   `--DESCRIPTION -> DESCRIPTION [0:13]
|       `--TEXT -> ://www.something.com [0:13]
`--EOF -> <EOF> [0:33]
```